lib/Sema/SemaChecking.cpp

```diff
@@ -10666,22 +10669,32 @@ void Sema::DiagnoseCheckedDecl(const DeclaratorDecl *Decl) {
     Ty = Field->getType();
   }
 
-  bool checkedScopeError = false;
+  bool Res = true;
   int TypeKind = 0;
   if (TargetDecl) {
     // type is unchecked pointer/array type w/o bounds-safe interface
     // bounds safe interface is always checked type
-    QualType InterOpTy = GetCheckedCInteropType(TargetDecl);
-    if ((Ty->isUncheckedPointerType() || Ty->isUncheckedArrayType()) &&
-        InterOpTy.isNull()) {
-      checkedScopeError = true;
-      TypeKind = (Ty->isUncheckedPointerType() ? 0 : 1);
+
+    if (isa<PointerType>(Ty) || isa<ArrayType>(Ty)) {
+      // check if type is checked pointer type or it has interop type
+      QualType InterOpTy = GetCheckedCInteropType(TargetDecl);
+      if (!Ty->hasCheckedType() && InterOpTy.isNull()) {
+        Res = false;
+        TypeKind = (Ty->isUncheckedPointerType() ? 0 : 1);
+      }
     }
   }
-  if (checkedScopeError) {
-    Diag(TargetDecl->getLocStart(), diag::err_checked_scope_type) << DeclKind
+  if (!Res) {
```

Comment:
Thanks for your suggestion
I consider how to handle error message for both use and declaration more efficiently.
If definition has an unchecked type error in a checked block, it produces error message for only declaration
Otherwise, it produces error message for each uses